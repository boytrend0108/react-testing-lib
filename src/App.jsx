import { useEffect, useState } from 'react';
import './App.css';

function App() {
  const [data, setData] = useState(false);
  const [toggle, setToggle] = useState(false);
  const [value, setValue] = useState('');

  function handleClick() {
    setToggle(!toggle);
  }

  useEffect(() => {
    setTimeout(() => {
      setData(true);
    }, 100);

    return () => {
      //clear timeout
    };
  }, []);

  return (
    <div className=''>
      {toggle && <div data-testid='toggle-div'>toogle</div>}
      {data && <p>data</p>}
      <h1 style={{ color: 'red' }}>Hello world!</h1>
      <h2 style={{ color: 'red' }} data-testid='value'>
        {value}
      </h2>
      <button data-testid='button1' onClick={handleClick}>
        click me
      </button>
      <input
        type='text'
        placeholder='enter name'
        onChange={(e) => setValue(e.target.value)}
      />
    </div>
  );
}

export default App;
