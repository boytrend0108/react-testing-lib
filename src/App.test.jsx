import { render, screen, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';
import App from './App';

describe('test App component', () => {
  test('loads and displays greeting', async () => {
    render(<App />);

    const title = screen.getByText(/Hello world!/i);
    const button = screen.getByTestId('button1');
    const input = screen.getByPlaceholderText(/enter name/i);

    screen.debug();
    expect(title).toBeInTheDocument();
    expect(button).toBeInTheDocument();
    expect(input).toBeInTheDocument();
    expect(input).toMatchSnapshot();
  });

  test('loads and displays greeting2', async () => {
    render(<App />);

    const title = screen.queryByText('some text');
    expect(title).toBeNull();
  });

  test('check style', async () => {
    render(<App />);

    const title = screen.queryByText(/Hello world!/i);
    expect(title).toHaveStyle({ color: 'red' });
  });

  test('check async ', async () => {
    render(<App />);

    const data = await screen.findByText(/data/i);
    expect(data).toBeInTheDocument();
    screen.debug();
  });

  test('check click on button', async () => {
    render(<App />);

    const btn = screen.getByTestId('button1');
    const toggleDiv = screen.queryByTestId('toggle-div');
    expect(toggleDiv).toBeNull();
    fireEvent.click(btn);
    expect(screen.queryByTestId('toggle-div')).toBeInTheDocument();
    fireEvent.click(btn);
    expect(toggleDiv).toBeNull();
  });

  test('check input', async () => {
    render(<App />);

    const input = screen.getByPlaceholderText(/enter name/i);
    const value = screen.getByTestId('value');
    expect(value).toContainHTML('');
    fireEvent.input(input, {
      target: { value: '123' },
    });
    expect(value).toContainHTML('123');
  });
});
