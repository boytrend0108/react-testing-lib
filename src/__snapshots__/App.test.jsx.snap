// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test App component loads and displays greeting 1`] = `
<input
  placeholder="enter name"
  type="text"
/>
`;
